### PR TITLE
Adopt bndtools.pde project from bnd tools

### PR DESCRIPTION
**Pull request by laeubi**
See https://bnd.discourse.group/t/announcement-bnd-and-pde-cooperation/372 this is a first step in migrating the things currently located under https://github.com/bndtools/bnd/tree/master/bndtools.pde.

As these are completely depending on PDE already they can be intercepted into the already existing pde-core without an problem for PDE/BNDtools users.

grafik

Corresponding bndtools change:
- https://github.com/bndtools/bnd/pull/5887

Requires:
- https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1538

### CODE REVIEW

**Comment by HannesWell** [comment]
Nice. I'm glad to see progress on this effort. 👍🏽 
Don't we need a CQ for this as well?

**Comment by laeubi** [comment]
I created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/11531

**Comment by iloveeclipse** [comment]
I see now errors in the IDE starting with this one:
`Neither an available bundle nor the associated JRE export package 'biz.aQute.resolve'

What is missing, which bundle is supposed to export that package and why it is not known by SDK?
Note: I'm using SDK build I20231128-1800.

**Comment by akurtakov** [comment]
@iloveeclipse Dependency has been added in target platform today https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1538 and since there is no build since then it's not included in the SDK.

File: ui/org.eclipse.pde.ui/plugin.xml [hunk 1]
```diff
@@ -2496,4 +2496,47 @@
...
+            icon="platform:/plugin/org.eclipse.pde.ui/icons/bndtools/database.png"
+            id="bndtools.RepositoryTargetLocationProvisioner"
+            name="Bnd Repository">
+         <description>
+            Download plug-ins from a repository defined in a Bnd workspace
+         </description>
+      </locationProvider>
+   </extension>
+   <extension
+         point="org.eclipse.pde.core.targetLocations"
```
Comment (iloveeclipse) [inline ui/org.eclipse.pde.ui/plugin.xml:2512]: This and below contribution show errors in the IDE:

```
Description	Resource	Path	Location	Type
Attribute 'canUpdate' of element 'targetLocation' must be defined.	plugin.xml	/org.eclipse.pde.ui	line 2515	Plug-in Problem
Attribute 'canUpdate' of element 'targetLocation' must be defined.	plugin.xml	/org.eclipse.pde.ui	line 2537	Plug-in Problem
```

Please fix either extension point or contribution here, not sure what was intended.

**Pull request merged**